public/app/features/transformers/calculateHeatmap/heatmap.ts

```diff
@@ -200,6 +256,11 @@ export function rowsToCellsHeatmap(opts: RowsHeatmapOptions): DataFrame {
     });
   }
 
+  // Only clear yOrdinalDisplay when using linear scale
+  if (useLinearScale) {
+    custom.yOrdinalDisplay = undefined;
+  }
+
```

Comment:
Clear yOrdinalDisplay for Linear scale
Switches rendering from ordinal to numeric mode